### PR TITLE
[ADF-4829] - Add e2e tests for custom action menu in TaskList and ProcessList

### PR DESCRIPTION
**Pull request by gmandakini**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts [hunk 1]
```diff
@@ -156,4 +158,29 @@ export class TaskListCloudComponentPage {
...
+    }
+
+    async rightClickOnRow(taskId: string): Promise<void> {
+        await this.dataTable.rightClickOnRow('Id', taskId);
+    }
+
+    async clickContextMenuActionNamed(actionName): Promise<void> {
+        await BrowserActions.clickExecuteScript(`button[data-automation-id="context-${actionName}"]`);
+    }
+
```
Comment (gmandakini) [inline lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts]: I see the above 5 methods duplicated across the task-list-cloud-component.page.ts and process-list-cloud-component.page.ts. Open to suggestions to make this code better?

Comment (cristinaj) [reply lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts]: I think clickOptionsButton can be moved to documentList or datatable PO. clickOnCustomActionMenu(), clickContextMenuActionNamed() and isCustomActionEnabled() can't be part of documentList or datatable, maybe can be part of another PO related to menu. rightClickOnRow() already uses a datatable method, so should stay here.

Comment (gmandakini) [reply lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts]: @eromano would you suggest to put the action menu methods in BrowserActions class? or maybe create a separate actionmenu class and add these methods there... pls advise.

File: e2e/pages/adf/demo-shell/process-services/tasksCloudDemoPage.ts [hunk 2]
```diff
@@ -157,15 +158,25 @@ export class TasksCloudDemoPage {
         await BrowserVisibility.waitUntilElementIsVisible(this.addActionTitle);
     }
 
-    async addAction(text: string): Promise<void> {
+    async addAction(text: string, disable: boolean): Promise<void> {
```
Comment (mauriziovitale) [inline e2e/pages/adf/demo-shell/process-services/tasksCloudDemoPage.ts]: this is a bad practice. A method should never have a boolean as input and switch from one behaviour to another based on the condition. Let's have a single responsibility and create 2 methods

File: lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts (hunk 1 above)
Comment (mauriziovitale) [reply lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts]: I don't see this as a big problem as long as we have only 2 places Task and Process

File: e2e/process-services-cloud/task-list-cloud-action-menu.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+        let processDefinitionService: ProcessDefinitionsService;
+        let processInstancesService: ProcessInstancesService;
+        let identityService: IdentityService;
+        let groupIdentityService: GroupIdentityService;
+        let queryService: QueryService;
+        let tasksService: TasksService;
+        let testUser, groupInfo, editProcess, deleteProcess, editTask, deleteTask;
+
+        const simpleApp = resources.ACTIVITI7_APPS.SIMPLE_APP.name;
```
Comment (mauriziovitale) [inline e2e/process-services-cloud/task-list-cloud-action-menu.e2e.ts]: can we use the same strategy as for the other parameters?
Basically add the resources as part of protractor params and use it as 
`browser.params. resources`

Comment (mauriziovitale) [reply e2e/process-services-cloud/task-list-cloud-action-menu.e2e.ts]: I will raise a different issue and refactor all the e2e

**Pull request merged**